kotlin.html.markdown

```diff
@@ -0,0 +1,401 @@
+// Testing if value is in range
+fun conditionalRange(x: Int, y: Int) {
+  if (x in 1..y-1)
+    print("OK")
+}
+
+fun conditionalIf(x: Int) {
+  // if branches can be blocks, and the last expression is the value of a block:
+  a = 9
+  b = 2
+  val max = if (a > b) { 
+    print("Choose a") 
+    a 
+  } 
+  else { 
+    print("Choose b") 
+    b 
+  }
+  
+  if (x > 0)
+    return x
+  else
+    return 0
+}
+
+fun conditionalWhen(obj: Any) {
+  when (obj) {
+    1          -> print("One")
+    "Hello"    -> print("Greeting")
+    is Long    -> print("Long")
+    !is String -> print("Not a string")
+    else       -> print("Unknown")
+  }
+}
+
+// _____________________________________________________________________ LOOPS
+
+fun loopDemo(args: Array<String>) {
+  print("for-each")
+  for (arg in args)
+    print(arg)
+
+  print("for i")
+  for (i in args.indices)
+    print(args[i])
+
+  print("while")
+  var i = 0
+  while (i < args.size())
+    print(args[i++])
+    
+  print("do while")
+  i = args.size() - 1
+  do {
+    println(args[i])
+  } while(i >= 0)
+  
+  // Break and continue are supported
+  var loopCount = 0
+  for (j in 0..20)
+    if (j.mod(2) == 0)
+      continue
+    else if(j == 11)
+      break
+    else
+      loopCount++
+  println("From 0 to 20, skipping even numbers and breaking at 11," +
+      " the loop counted $loopCount times.")
+  
+}
+
+// _________________________________________________________ RETURNS AND jUMPS
+
+/**
+ * Any expression in Kotlin may be marked with a label. Labels have the form 
+ * of the @ sign followed by an identifier, for example: @abc, @fooBar
+ */
+fun breakWitLabels() {
+  @loop for (i in 1..100) {
+    for (j in 1..100) {
+      if (...)
+        // jumps to the execution point right after the loop marked with the 
+        // label. A continue would instead proceeds to the next iteration of 
+        // the loop
+        break@loop
+    }
+  }
+  
+}
+
+/**
+ * @TODO [Return at Labels](http://kotlinlang.org/docs/reference/returns.html)
+ */
+
+// ____________________________________ NULL SAFETY AND AUTOMATIC TYPE CASTING
+
+// References are non-null by default. If otherwise, they must be marked.
+
+// Type checking and automatic casting. Notice the `Any` type.
+fun nullChecking0(obj: Any): Int? { // Because of `?`, null may be returned.
+  if (obj is String) {
+    // `obj` is automatically cast to `String` in this branch
+    return obj.length
+  }
+  
+  // else returns null
+}
+
+// Type checking and automatic casting, version2
+fun nullChecking1(obj: Any): Int? {
+  if (obj !is String)
+    return null
+    
+  // Automatic cast to `String`! we may use obj.length now.
+  return obj.length
+}
+
+// Automatic cast for right-hand side of `&&`
+fun nullChecking2(obj: Any): Int? {
+  if (obj is String && obj.length > 0)
+    return obj.length
+}
+
+// *If not null* shorthand 
+fun nullChecking3(obj: Any): Int? {
```

Comment:
Here and in the following two functions, obj should be Any?, otherwise the null-safety doesn't make sense